daemon/images/images.go

```diff
@@ -38,6 +38,20 @@ func (i *ImageService) Map() map[image.ID]*image.Image {
 	return i.imageStore.Map()
 }
 
+// rangeImageLayerChainIDs calls f sequentially for chain ID of each layer image consists of.
+// If f returns false, range stops the iteration.
+func rangeImageLayerChainIDs(img *image.Image, f func(layer.ChainID) bool) {
```

Comment:
`func rangeImageLayerChainIDs(img *image.Image, f func(layer.ChainID) error) error` you can bubble up the error to be more precise

Reply:
Hmm, after taking a closer look at this, I am not sure if returning an error in the callback is a cleaner solution here.

We have 2 usages here:
1. https://github.com/moby/moby/pull/42578/files?w=1#diff-b62ea5c3b3e9f7f294e285690608fbd87d02686b780e1356f4c31652a9542fe7R246-R248 which iterates over all chain IDs *always*
2. https://github.com/moby/moby/pull/42578/files?w=1#diff-b62ea5c3b3e9f7f294e285690608fbd87d02686b780e1356f4c31652a9542fe7R258-R274, which iterates over all chain IDs *unless* an error is encountered in one of the iterations

If we want to return an error in the callback, then discarding it seems wrong - at least that would not be "idiomatic", so if we want to return an error, I think we would need to propagate it up to the caller.

Now if we refactor it in this way, we get a marginal improvement in usage 2., since now we could wrap the function call into a shortened if as in `if err := range...; err != nil` and avoid one assignment to `err` from within the callback. For the usage 1. that would make it more confusing though, in my opinion: the iteration never errors, but now we'd need to either discard the error value returned by the function or check for `!= nil`, which we know would never happen, which is IMO worse than a single assignment to `err` from within the callback in usage 2.

Do you agree?